src/features/api-keys/hooks.ts

```diff
@@ -0,0 +1,9 @@
+import { hooks } from '@balena/pinejs';
+
+hooks.addPureHook('POST', 'resin', 'api_keys', {
+	PRERUN: async ({ request }) => {
+		if (!request.values.provisioningKeyExpiry) {
+			// should we set a default expiry date here
```

Comment:
In `pineJS` this is set to `NULL` for existing entries: https://github.com/balena-io/pinejs/pull/476